slsDetectorServers/slsDetectorServer/src/slsDetectorServer_funcs.c

```diff
@@ -6329,35 +6328,37 @@ int get_inject_channel(int file_des) {
 int set_veto_photon(int file_des) {
     ret = OK;
     memset(mess, 0, sizeof(mess));
-    int args[3] = {-1, -1, -1};
 
+    int args[2] = {-1, -1};
     if (receiveData(file_des, args, sizeof(args), INT32) < 0)
         return printSocketReadError();
-    int values[args[2]];
+    int chipIndex = args[0];
+    int numChannels = args[1];
+
+    int gainIndices[args[1]];
```

Comment:
How large is the gainIndicies? I'm generally not so keen on VLA... We don't know the size of this before? Or if it's rather large maybe heap allocation is better?

Reply:
Its not large, and the size is not known at compilation (numchannels depends on detector). More for extra validation between client and server constants. Adding const in this case to not make it VLA.